src/extension/background-script/actions/accounts/add.ts

```diff
@@ -31,7 +17,10 @@ const add = async (
 
   const accountId = uuidv4();
   newAccount.config = encryptData(newAccount.config, password);
-  accounts[accountId] = newAccount;
+  accounts[accountId] = {
+    id: accountId,
+    ...newAccount,
```

Comment:
potentially ...newAccount could overwrite id. Maybe switch the order?

Reply:
Good point. According to TS that can not happen, but it's good to start with the spread anyways, thanks!